cli/command/container/utils.go

```diff
@@ -53,27 +75,20 @@ func waitExitOrRemoved(ctx context.Context, dockerCli *command.DockerCli, contai
 		return false
 	}
 
-	// Get events via Events API
-	f := filters.NewArgs()
-	f.Add("type", "container")
-	f.Add("container", containerID)
-	options := types.EventsOptions{
-		Filters: f,
-	}
-
-	eventCtx, cancel := context.WithCancel(ctx)
-	eventq, errq := dockerCli.Client().Events(eventCtx, options)
-
 	go func() {
 		defer cancel()
 
 		for {
 			select {
+			case <-eventCtx.Done():
+				if removeErr != nil {
+					logrus.Errorf("error removing container: %v", removeErr)
+					return
+				}
```

Comment:
Hmm... If container remove fails, removeErr will not be nil, the eventCtx will be cancelled, and the select will hit `eventCtx.Done()` and return.

If container remove does not fail then processing keeps going as usual until it gets the remove event and returns.